### PR TITLE
Update documents

### PR DESCRIPTION
**Pull request by Chuhaa**
## Purpose

- Update documents
- Add comments for client and listener `init`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,390 +1,51 @@
...
-### Pull the Module
-You can pull the Amazon S3 client from Ballerina Central:
-```shell
-$ bal pull ballerinax/aws.s3
-```
+[Amazon Simple Storage Service (Amazon S3)](https://aws.amazon.com/s3/) is an object storage service that offers industry-leading scalability, data availability, security, and performance.This means customers of all sizes and industries can use it to store and protect any amount of data for a range of use cases, such as data lakes, websites, mobile applications, backup and restore, archive, enterprise applications, IoT devices, and big data analytics.
+
+The module provides the capability to manage buckets and objects in AWS S3.
+For more information about configuration and operations, go to the module.
+- [ballerinax/aws.s3](https://docs.central.ballerina.io/ballerinax/aws.s3/1.0.0)
```
Comment (LakshanSS) [inline README.md:10]: Dont add version in the link
Please change link to latest agreed format

File: s3/Package.md [hunk 2]
```diff
@@ -1,333 +1,21 @@
...
-    if (deleteBucketResponse is error) {
-        log:printError("Error: " + deleteBucketResponse.toString());
-    } else {
-        log:printInfo("Successfully deleted bucket");
-    }
-}
-```
\ No newline at end of file
+## Useful links
+- Discuss code changes of the Ballerina project in [ballerina-dev@googlegroups.com](mailto:ballerina-dev@googlegroups.com).
```
Comment (RukshiW) [inline s3/Package.md:19]: Replace "in" with "via".

File: s3/Module.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+## Overview
+The module provides the capability to manage buckets and objects in [AWS S3](https://aws.amazon.com/s3/).
+
+This module supports [Amazon S3 REST API](https://docs.aws.amazon.com/AmazonS3/latest/API/Welcome.html) `2006-03-01` version.
+ 
+## Prerequisites
+Before using this connector in your Ballerina application, complete the following:
+- Create [AWS account](https://portal.aws.amazon.com/billing/signup?nc2=h_ct&src=default&redirect_url=https%3A%2F%2Faws.amazon.com%2Fregistration-confirmation#/start)
+- Obtain tokens - Follow [this link](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html)
```
Comment (RukshiW) [inline s3/Module.md:9]: Suggestion: "[Obtain Tokens](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html)."

Make "Obtain Tokens" the link text as shown above. It is OK if the whole bullet point is link text.

File: s3/Module.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+To use the AWS S3 connector in your Ballerina application, update the .bal file as follows:
+
+### Step 1: Import connector
+First, import the `ballerinax/aws.s3` module into the Ballerina project.
+```ballerina
+import ballerinax/aws.s3;
+```
+
+### Step 2: Create a new connector instance
+Create client using connection configuration.
```
Comment (RukshiW) [inline s3/Module.md:21]: Is this sentence needed? Doesn't the next sentence cover what this says?

File: s3/Module.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    string bucketName = "name";
+
+    public function main() returns error? {
+        _ = check amazonS3Client->createBucket(bucketName);
+    }
+    ```
+2. Use `bal run` command to compile and run the Ballerina program.
+
+## Quick reference
+Code snippets of some frequently used functions:
```
Comment (RukshiW) [inline s3/Module.md:47]: Suggestion: "The following are code snippets of some frequently used functions:"

Comment (Chuhaa) [reply s3/Module.md]: We decided to remove quick reference.

File: s3/Module.md (hunk 4 above)
Comment (Chuhaa) [reply s3/Module.md]: No. Not needed. Fixed in https://github.com/ballerina-platform/module-ballerinax-aws.s3/pull/55/commits/df2a25cd27ea5b50683e8473e083cb5f4a90927b

File: s3/Module.md (hunk 3 above)
Comment (Chuhaa) [reply s3/Module.md]: Fixed in https://github.com/ballerina-platform/module-ballerinax-aws.s3/pull/55/commits/df2a25cd27ea5b50683e8473e083cb5f4a90927b

File: s3/Package.md (hunk 2 above)
Comment (Chuhaa) [reply s3/Package.md]: Fixed in https://github.com/ballerina-platform/module-ballerinax-aws.s3/pull/55/commits/df2a25cd27ea5b50683e8473e083cb5f4a90927b

File: s3/Module.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
+## Overview
+The module provides the capability to manage buckets and objects in [AWS S3](https://aws.amazon.com/s3/).
+
+This module supports [Amazon S3 REST API](https://docs.aws.amazon.com/AmazonS3/latest/API/Welcome.html) `2006-03-01` version.
+ 
+## Prerequisites
+Before using this connector in your Ballerina application, complete the following:
+- Create [AWS account](https://portal.aws.amazon.com/billing/signup?nc2=h_ct&src=default&redirect_url=https%3A%2F%2Faws.amazon.com%2Fregistration-confirmation#/start)
```
[the remark this answers is not in the document]
Comment (Chuhaa) [reply s3/Module.md]: Fixed in https://github.com/ballerina-platform/module-ballerinax-aws.s3/pull/55/commits/df2a25cd27ea5b50683e8473e083cb5f4a90927b

File: README.md [hunk 7]
```diff
@@ -1,390 +1,51 @@
...
-
-* Discuss the code changes of the Ballerina project in [ballerina-dev@googlegroups.com](mailto:ballerina-dev@googlegroups.com).
+ 
+## Code of conduct
+ 
+All contributors are encouraged to read the [Ballerina Code of Conduct](https://ballerina.io/code-of-conduct).
+ 
+## Useful links
+ 
+* Discuss code changes of the Ballerina project in [ballerina-dev@googlegroups.com](mailto:ballerina-dev@googlegroups.com).
```
[the remark this answers is not in the document]
Comment (Chuhaa) [reply README.md]: Fixed in https://github.com/ballerina-platform/module-ballerinax-aws.s3/pull/55/commits/df2a25cd27ea5b50683e8473e083cb5f4a90927b

File: README.md [hunk 8]
```diff
@@ -1,390 +1,51 @@
...
+   bal build -c --skip-tests ./s3
+   ```
 ## Contributing to Ballerina
-
-As an open source project, Ballerina welcomes contributions from the community. 
-
+ 
+As an open source project, Ballerina welcomes contributions from the community.
+ 
 For more information, go to the [contribution guidelines](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md).
```
[the remark this answers is not in the document]
Comment (Chuhaa) [reply README.md]: Fixed in https://github.com/ballerina-platform/module-ballerinax-aws.s3/pull/55/commits/df2a25cd27ea5b50683e8473e083cb5f4a90927b

File: README.md [hunk 9]
```diff
@@ -1,390 +1,51 @@
...
+  * [OpenJDK](https://adoptopenjdk.net/)
+ 
+       > **Note:** Set the JAVA_HOME environment variable to the path name of the directory into which you installed
+       JDK.
+ 
+2. Download and install [Ballerina Swan Lake Beta2](https://ballerina.io/)
+
+### Building the source
+ 
+Execute the commands below to build from the source.
```
[the remark this answers is not in the document]
Comment (Chuhaa) [reply README.md]: Fixed in https://github.com/ballerina-platform/module-ballerinax-aws.s3/pull/55/commits/df2a25cd27ea5b50683e8473e083cb5f4a90927b

File: README.md [hunk 10]
```diff
@@ -1,390 +1,51 @@
...
-    ```shell
-    $ bal build
-    ```
-
-3. To build the module without the tests:
-    ```shell 
-    $ bal build --skip-tests
-    ```
-
+[Amazon Simple Storage Service (Amazon S3)](https://aws.amazon.com/s3/) is an object storage service that offers industry-leading scalability, data availability, security, and performance.This means customers of all sizes and industries can use it to store and protect any amount of data for a range of use cases, such as data lakes, websites, mobile applications, backup and restore, archive, enterprise applications, IoT devices, and big data analytics.
```
[the remark this answers is not in the document]
Comment (Chuhaa) [reply README.md]: Fixed in https://github.com/ballerina-platform/module-ballerinax-aws.s3/pull/55/commits/df2a25cd27ea5b50683e8473e083cb5f4a90927b

File: README.md (hunk 1 above)
Comment (Chuhaa) [reply README.md]: Fixed.

**Review by abeykoon — APPROVED** [review]

File: s3/client.bal [hunk 11]
```diff
@@ -263,23 +263,23 @@ isolated function setDefaultHeaders(string amazonHost) returns map<string> {
     return requestHeaders;
 }
 
-# Verify the existence of credentials.
+# Verifies the existence of credentials.
 #
-# + accessKeyId - The access key is of the Amazon S3 account.
-# + secretAccessKey - The secret access key of the Amazon S3 account.
+# + accessKeyId - The access key is of the Amazon S3 account
```
Comment (RukshiW) [inline s3/client.bal:268]: Is the "is" needed?

File: s3/client.bal [hunk 12]
```diff
@@ -263,23 +263,23 @@ isolated function setDefaultHeaders(string amazonHost) returns map<string> {
...
     if ((accessKeyId == "") || (secretAccessKey == "")) {
         return error(EMPTY_VALUES_FOR_CREDENTIALS_ERROR_MSG);
     }
 }
 
-# AmazonS3 Connector configurations can be setup here.
-# + accessKeyId - The access key is of the Amazon S3 account.
-# + secretAccessKey - The secret access key of the Amazon S3 account.
+# AmazonS3 Connector configurations.
+# + accessKeyId - The access key is of the Amazon S3 account
```
Comment (RukshiW) [inline s3/client.bal:279]: Is the "is" needed?

**Pull request merged**